src/main/java/org/opensearch/securityanalytics/threatIntel/iocscan/dao/IocMatchService.java

```diff
@@ -152,4 +167,44 @@ public void createIndexIfNotExists(final ActionListener<Void> listener) {
             listener.onFailure(e);
         }
     }
+
+    public void searchIocMatches(SearchSourceBuilder searchSourceBuilder, final ActionListener<GetIocFindingsResponse> actionListener) {
+        SearchRequest searchRequest = new SearchRequest()
+                .source(searchSourceBuilder)
+                .indices(INDEX_NAME);
+
+        client.search(searchRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(SearchResponse searchResponse) {
+                try {
+                    long totalIocFindingsCount = searchResponse.getHits().getTotalHits().value;
+                    MultiGetRequest mGetRequest = new MultiGetRequest();
+                    List<IocMatchWithDocs> iocMatchWithDocs = new ArrayList<>();
+                    List<IocMatch> iocMatches = new ArrayList<>();
+
+                    for (SearchHit hit: searchResponse.getHits()) {
+                        XContentParser xcp = XContentType.JSON.xContent()
+                                .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, hit.getSourceAsString());
+                        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp);
+                        IocMatch iocMatch = IocMatch.parse(xcp);
+                        iocMatches.add(iocMatch);
+
+                        List<String> documentIds = iocMatch.getRelatedDocIds();
+                    }
+
+                    for (IocMatch iocMatch: iocMatches) {
+                        iocMatchWithDocs.add(new IocMatchWithDocs(iocMatch, List.of()));
+                    }
+                    actionListener.onResponse(new GetIocFindingsResponse((int) totalIocFindingsCount, iocMatchWithDocs));
+                } catch (Exception ex) {
+                    this.onFailure(ex);
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
```

Comment:
return empty list for indexnotfound exception

Reply:
fixed it.